staging/src/k8s.io/apiserver/pkg/authorization/cel/compile.go

```diff
@@ -20,22 +20,32 @@ import (
 	"fmt"
 
 	"github.com/google/cel-go/cel"
+	celast "github.com/google/cel-go/common/ast"
+	"github.com/google/cel-go/common/operators"
 	"github.com/google/cel-go/common/types/ref"
 
 	authorizationv1 "k8s.io/api/authorization/v1"
 	"k8s.io/apimachinery/pkg/util/version"
 	apiservercel "k8s.io/apiserver/pkg/cel"
 	"k8s.io/apiserver/pkg/cel/environment"
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 )
 
 const (
 	subjectAccessReviewRequestVarName = "request"
+
+	fieldSelectorVarName = "fieldSelector"
+	labelSelectorVarName = "labelSelector"
 )
 
 // CompilationResult represents a compiled authorization cel expression.
 type CompilationResult struct {
 	Program            cel.Program
 	ExpressionAccessor ExpressionAccessor
+
+	UsesFieldSelector bool
+	UsesLabelSelector bool
```

Comment:
I'm not entirely clear on why we need this.  Doesn't the parse select the features based on the version of CEL compatibility requested?

Reply:
it's much more complex/expensive to transform the selectors into the unstructured object the matchConditions operate on than all the other request attributes, so if the expressions aren't using those fields, I skip populating the variables for CEL to evaluate on

Reply:
I'll add a comment explaining this